registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/eventbus/VertxMosipEventBus.java

```diff
@@ -121,4 +126,44 @@ public void send(MessageBusAddress toAddress, MessageDTO message) {
 		this.vertx.eventBus().send(messageBusAddress.getAddress(), jsonObject.toString());
 	}
 
+	@Override
+	public void consumerHealthCheck(EventHandler<EventDTO, Handler<AsyncResult<HealthCheckDTO>>> eventHandler,String address) {
+		Boolean isRegistered = vertx.eventBus().consumer(address).isRegistered();
+		EventDTO eventDTO = new EventDTO();
+		HealthCheckDTO healthCheckDTO = new HealthCheckDTO();
+		healthCheckDTO.setEventBusConnected(isRegistered);
+		try {
+			eventDTO.setBody(new JsonObject(JsonUtil.objectMapperObjectToJson(healthCheckDTO)));
+		} catch (IOException e) {
+			healthCheckDTO.setEventBusConnected(false);
+			healthCheckDTO.setFailureReason(e.getMessage());
+			eventHandler.handle(eventDTO, res -> {
+			});
+		}
+		eventHandler.handle(eventDTO, res -> {
+
+		});
+		
+	}
+
+	@Override
+	public void sendHealthCheck(EventHandler<EventDTO, Handler<AsyncResult<HealthCheckDTO>>> eventHandler,
+			String address) {
+		vertx.eventBus().send(address, HealthConstant.PING);

```

Comment:
there is no boolean or exception handling for this ?

Reply:
Exception handled